plasmapy/particles/nuclear.py

```diff
@@ -11,6 +13,263 @@
+    Fusion Cross-Sections and Thermal Reactivities.
+    Nuclear Fusion 1992, 32 (4).
+    https://iopscience.iop.org/article/10.1088/0029-5515/32/4/I07/meta
+    """
+
+    def __init__(self, Bg, A, B, energy_range, error_max):
+        """
+        Parameters
+        ----------
+        Bg : float
+            \sqrt{keV}
+        A : list
+            Pade numerator coefficients
+        B : list
+            Pade denominator coefficients
+        energy_range : list with two elements
+            [lo, hi] in keV
+            Valid energy range for the formula
+        error_max : in percent
+            Maximum error of the formula from the best known cross section data
+        """
+        self.Bg = Bg # in \sqrt{keV}
+        self.A1, self.A2, self.A3, self.A4, self.A5 = A
+        self.B1, self.B2, self.B3, self.B4 = B
+        self.energy_range = energy_range
+        self.error_max = error_max
+
+    def sigma(self, energy) -> u.Quantity:
+        """Cross section
+
+        Parameters
+        ----------
+        Energy: `~astropy.units.Quantity`
+            Center-of-mass-frame energy in keV
+
+        Returns
+        -------
+        sigma : `~astropy.units.Quantity`
+            The reaction cross section in millibarns
+
+        Notes
+        -----
+        See Equations (8) and (9) of Bosch & Hale
+        """
+        # TODO return warning if out of energy range?
+        E = energy.to('keV').value
+        numerator = self.A1 + E * (self.A2 + E * (self.A3 + E * (self.A4 + E * self.A5)))
+        denominator = 1 + E * (self.B1 + E * (self.B2 + E * (self.B3 + E * self.B4)))
+        S = numerator / denominator
+        sigma = S / (E * np.exp(self.Bg / np.sqrt(E)))
+        return sigma * u.millibarn
+
+class BoschHaleRateCoefficientFit():
+    """
+    Evaluates nuclear reaction rate coefficients in the form of the fit given in
+    the referenced paper [1], Equations (12-14)
+
+    References
+    ----------
+    [1] Bosch, H.-S.; Hale, G. M. Improved Formulas for
+    Fusion Cross-Sections and Thermal Reactivities.
+    Nuclear Fusion 1992, 32 (4).
+    https://iopscience.iop.org/article/10.1088/0029-5515/32/4/I07/meta
+    """
+
+    def __init__(self, Bg, mrcsq, C, energy_range, error_max):
+        """
+        Parameters
+        ----------
+        Bg : float
+            \sqrt{keV}
+        mrcsq : float
+            keV, an energy
+        C : list
+            Coefficients
+        energy_range : list with two elements
+            keV, [low, high]
+            Valid energy range for the formula
+        error_max : in percent
+            Maximum error of the formula from the best known cross section data
+        """
+        self.Bg = Bg
+        self.mrcsq = mrcsq
+        self.C = C
+        self.energy_range = energy_range
+        self.error_max = error_max
+
+    def sigma_v(self, temperature):
+        # TODO return warning if out of energy range?
+        T = temperature.to(u.keV).value
```

Comment:
We'll have to be careful here since keV are units of energy rather than temperature, which makes it more complicated to use `astropy.units`.  It might be necessary to use the `u.temperature_energy(https://docs.astropy.org/en/stable/units/equivalencies.html?highlight=equivalencies#temperature-energy-equivalency)` equivalency here, though this might also be able to be taken care of with `@plasmapy.utils.validate_quantities`.